src/odemis/gui/cont/tabs.py

```diff
@@ -4389,7 +4393,481 @@ def _on_align_move_done(self, f=None):
+
+        :param shift (dict --> float): Relative movement of the beam shift with the axis as keys (x/y)
+        """
+        beamShiftVA = self._sem_stream.emitter.beamShift
+        try:
+            if "x" in shift:
+                beamShiftVA.value = (beamShiftVA.value[0] + shift["x"], beamShiftVA.value[1])
+        except IndexError:
+            show_message(wx.GetApp().main_frame,
+                         "Reached the limits of the beam shift, cannot move any further in x direction.",
+                         timeout=3.0, level=logging.WARNING)
+            logging.error("Reached the limits of the beam shift, cannot move any further in x direction.")
+
+        try:
+            if "y" in shift:
+                beamShiftVA.value = (beamShiftVA.value[0], beamShiftVA.value[1] + shift["y"])
+        except IndexError:
+            show_message(wx.GetApp().main_frame,
+                         "Reached the limits of the beam shift, cannot move any further in y direction.",
+                         timeout=3.0, level=logging.WARNING)
+            logging.error("Reached the limits of the beam shift, cannot move any further in y direction.")
+
+        updated_stage_pos = self._stage.getMetadata()[model.MD_FAV_POS_ACTIVE]
+        updated_stage_pos.update({"x": self._stage.position.value["x"], "y": self._stage.position.value["y"]})
+        self._stage.updateMetadata({model.MD_FAV_POS_ACTIVE: updated_stage_pos})
+
+    def _set_flm_alignment_mode(self):
+        """
+        Sets the FLM alignment mode with the correct streams, stream controllers, instructions and alignment controls.
+        """
+        self.stream_bar_controller.pauseAllStreams()
+        self._set_top_and_bottom_stream_and_settings(self._FLM_view_and_control,
+                                                     self._SEM_view_and_control)
+
+        doc_path = pkg_resources.resource_filename("odemis.gui", "doc/enzel_flm_alignment.html")
+        self.panel.html_alignment_doc.LoadPage(doc_path)
+
+        self.panel.controls_step_size_slider.SetRange(100e-9, 50e-6)
+        self.panel.controls_step_size_slider.set_position_value(10e-6)
+
+        self.panel.pnl_z_align_controls.Show(False)
+        self.panel.pnl_sem_align_controls.Show(False)
+        self.panel.pnl_flm_align_controls.Show(True)
+        fix_static_text_clipping(self.panel)
+
+    def _flm_alignment_controls(self, axis, proportion, evt):
+        """
+        Links the FLM alignments mode control buttons to the stage and updates the correct metadata..
+
+        :param axis (str): axis which is controlled by the button.
+        :param proportion (int): direction of the movement represented by the button (-1/1)
+        :param evt (GenButtonEvent): Clicked event
+        """
+        # Only proceed if there is no currently running target_position
+        if hasattr(self, "_flm_move_future") and self._flm_move_future._state == RUNNING:
```

Comment:
You could define `_flm_move_future` and its similar others in `_init_` with None, that way you don't need to check its existence every time.

Reply:
Then it would actually crash. Currently the if statement assumes the future is containing a real future and None would fail the code. I could check then if the attribute is defined as a future and not as None but that wouldn't decrease lines of code.

Reply:
Right, you can do `if self._flm_move_future` but it won't decrease much..